configure --with-target= fails to install (sed cannot find prefixed scripts)

### What version of hwloc are you using?

2.7.0

### Which operating system and hardware are you running on?

Linux <COMPUTER_NAME> 4.4.0-19041-Microsoft #2311-Microsoft Tue Nov 08 17:09:00 PST 2022 x86_64 x86_64 x86_64 GNU/Linux

Machine (16GB total) + Package L#0
  NUMANode L#0 (P#0 16GB)
  Core L#0 + PU L#0 (P#0)
  Core L#1 + PU L#1 (P#1)

My computer runs Windows 10 Enterprise (WSL), with an Intel Pentium G3258, 2*8GB Patriot Viper DDR3-1600 RAM, and built-in graphics.

### Details of the problem

I am trying to build Hwloc. `./configure --prefix="/root/MiOSR/toolchain/i686-elf/" --target=i686-elf` works, and so does `make -s`. But when I try to run `make -s check`, test-hwloc-compress-dir.sh fails to run.

### Additional information

I tried to fix a previous error that was related to `gunzip` and `tar` failing to unzip `test-hwloc-compress-dir.input.tar.gz` and `test-hwloc-compress-dir.output.tar.gz`. So, I tried to unzip them manually.

Here's the full `test-suite.log`:
```
=============================================
   hwloc 2.9.0: utils/hwloc/test-suite.log
=============================================

# TOTAL: 8
# PASS:  7
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: test-hwloc-compress-dir.sh
================================

ls: cannot access '/tmp/foosawUN6/test-hwloc-compress-dir.input': No such file or directory
Compressed 0 new topologies (0 were already compressed)
Kept 0 new topologies non-compressed (0 were already non-compressed)
/usr/bin/diff: /tmp/foosawUN6/test-hwloc-compress-dir.output: No such file or directory
FAIL test-hwloc-compress-dir.sh (exit status: 2)
```

huh, now that I run `make -s install`, it doesn't work
here's the full log:
```
Making install in .
make[1]: Entering directory '/root/Downloads/hwloc-2.9.0/utils/hwloc'
make[2]: Entering directory '/root/Downloads/hwloc-2.9.0/utils/hwloc'
 /usr/bin/mkdir -p '/root/MiOSR/toolchain/i686-elf/bin'
  /bin/bash ../../libtool   --mode=install /usr/bin/install -c hwloc-annotate '/root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-annotate'
libtool: install: /usr/bin/install -c .libs/hwloc-annotate /root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-annotate
  /bin/bash ../../libtool   --mode=install /usr/bin/install -c hwloc-bind '/root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-bind'
libtool: install: /usr/bin/install -c .libs/hwloc-bind /root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-bind
  /bin/bash ../../libtool   --mode=install /usr/bin/install -c hwloc-calc '/root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-calc'
libtool: install: /usr/bin/install -c .libs/hwloc-calc /root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-calc
  /bin/bash ../../libtool   --mode=install /usr/bin/install -c hwloc-diff '/root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-diff'
libtool: install: /usr/bin/install -c .libs/hwloc-diff /root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-diff
  /bin/bash ../../libtool   --mode=install /usr/bin/install -c hwloc-distrib '/root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-distrib'
libtool: install: /usr/bin/install -c .libs/hwloc-distrib /root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-distrib
  /bin/bash ../../libtool   --mode=install /usr/bin/install -c hwloc-info '/root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-info'
libtool: install: /usr/bin/install -c .libs/hwloc-info /root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-info
  /bin/bash ../../libtool   --mode=install /usr/bin/install -c hwloc-patch '/root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-patch'
libtool: install: /usr/bin/install -c .libs/hwloc-patch /root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-patch
  /bin/bash ../../libtool   --mode=install /usr/bin/install -c hwloc-ps '/root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-ps'
libtool: install: /usr/bin/install -c .libs/hwloc-ps /root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-ps
  /bin/bash ../../libtool   --mode=install /usr/bin/install -c hwloc-gather-cpuid '/root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-gather-cpuid'
libtool: install: /usr/bin/install -c .libs/hwloc-gather-cpuid /root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-gather-cpuid
 /usr/bin/mkdir -p '/root/MiOSR/toolchain/i686-elf/bin'
 /usr/bin/install -c hwloc-compress-dir '/root/MiOSR/toolchain/i686-elf/bin/./i686-elf-hwloc-compress-dir'
make  install-exec-hook
make[3]: Entering directory '/root/Downloads/hwloc-2.9.0/utils/hwloc'
/usr/bin/sed -e 's/HWLOC_top_builddir\/utils\/hwloc/bindir/' -e 's/HWLOC_top_builddir\/utils\/lstopo/bindir/' -e '/HWLOC_top_builddir/d' /root/MiOSR/toolchain/i686-elf/bin/hwloc-compress-dir > /root/MiOSR/toolchain/i686-elf/bin/hwloc-compress-dir.tmp && mv -f /root/MiOSR/toolchain/i686-elf/bin/hwloc-compress-dir.tmp /root/MiOSR/toolchain/i686-elf/bin/hwloc-compress-dir
/usr/bin/sed: can't read /root/MiOSR/toolchain/i686-elf/bin/hwloc-compress-dir: No such file or directory
make[3]: *** [Makefile:1829: install-exec-hook] Error 2
make[3]: Leaving directory '/root/Downloads/hwloc-2.9.0/utils/hwloc'
make[2]: *** [Makefile:1727: install-exec-am] Error 2
make[2]: Leaving directory '/root/Downloads/hwloc-2.9.0/utils/hwloc'
make[1]: *** [Makefile:1657: install-am] Error 2
make[1]: Leaving directory '/root/Downloads/hwloc-2.9.0/utils/hwloc'
make: *** [Makefile:1273: install-recursive] Error 1
```

--- a/utils/hwloc/Makefile.am
+++ b/utils/hwloc/Makefile.am
@@ -1,6 +1,7 @@
 # Copyright © 2009-2021 Inria.  All rights reserved.
 # Copyright © 2009-2012, 2014, 2016 Université Bordeaux
 # Copyright © 2009-2014 Cisco Systems, Inc.  All rights reserved.
+# Copyright © 2023      Université de Reims Champagne-Ardenne.  All rights reserved.
 #
 # See COPYING in top-level directory.
 
@@ -160,12 +161,16 @@ endif HWLOC_HAVE_LINUX
 	@ $(SEDMAN) \
 	  > $@ < $<
 
+
+hwloc_compress_dir_name=`echo "hwloc-compress-dir" | $(SED) -e "$(transform)"`
+hwloc_gather_topology_name=`echo "hwloc-gather-topology" | $(SED) -e "$(transform)"`
+
 install-exec-hook:
-	$(SED) -e 's/HWLOC_top_builddir\/utils\/hwloc/bindir/' -e 's/HWLOC_top_builddir\/utils\/lstopo/bindir/' -e '/HWLOC_top_builddir/d' $(DESTDIR)$(bindir)/hwloc-compress-dir > $(DESTDIR)$(bindir)/hwloc-compress-dir.tmp && mv -f $(DESTDIR)$(bindir)/hwloc-compress-dir.tmp $(DESTDIR)$(bindir)/hwloc-compress-dir
-	chmod +x $(DESTDIR)$(bindir)/hwloc-compress-dir
+	$(SED) -e 's/HWLOC_top_builddir\/utils\/hwloc/bindir/' -e 's/HWLOC_top_builddir\/utils\/lstopo/bindir/' -e '/HWLOC_top_builddir/d' $(DESTDIR)$(bindir)/$(hwloc_compress_dir_name) > $(DESTDIR)$(bindir)/$(hwloc_compress_dir_name).tmp && mv -f $(DESTDIR)$(bindir)/$(hwloc_compress_dir_name).tmp $(DESTDIR)$(bindir)/$(hwloc_compress_dir_name)
+	chmod +x $(DESTDIR)$(bindir)/$(hwloc_compress_dir_name)
 if HWLOC_HAVE_LINUX
-	$(SED) -e 's/HWLOC_top_builddir\/utils\/lstopo/bindir/' -e 's/HWLOC_top_builddir\/utils\/hwloc/bindir/' -e '/HWLOC_top_builddir/d' $(DESTDIR)$(bindir)/hwloc-gather-topology > $(DESTDIR)$(bindir)/hwloc-gather-topology.tmp && mv -f $(DESTDIR)$(bindir)/hwloc-gather-topology.tmp $(DESTDIR)$(bindir)/hwloc-gather-topology
-	chmod +x $(DESTDIR)$(bindir)/hwloc-gather-topology
+	$(SED) -e 's/HWLOC_top_builddir\/utils\/lstopo/bindir/' -e 's/HWLOC_top_builddir\/utils\/hwloc/bindir/' -e '/HWLOC_top_builddir/d' $(DESTDIR)$(bindir)/$(hwloc_gather_topology_name) > $(DESTDIR)$(bindir)/$(hwloc_gather_topology_name).tmp && mv -f $(DESTDIR)$(bindir)/$(hwloc_gather_topology_name).tmp $(DESTDIR)$(bindir)/$(hwloc_gather_topology_name)
+	chmod +x $(DESTDIR)$(bindir)/$(hwloc_gather_topology_name)
 endif HWLOC_HAVE_LINUX
 
 distclean-local:

--- a/utils/lstopo/Makefile.am
+++ b/utils/lstopo/Makefile.am
@@ -2,6 +2,7 @@
 # Copyright © 2009-2012, 2014 Université Bordeaux
 # Copyright © 2009-2014 Cisco Systems, Inc.  All rights reserved.
 # Copyright © 2020 Hewlett Packard Enterprise.  All rights reserved.
+# Copyright © 2023      Université de Reims Champagne-Ardenne.  All rights reserved.
 #
 # See COPYING in top-level directory.
 
@@ -89,30 +90,34 @@ SEDMAN = $(SED) -e 's/%PACKAGE_NAME%/@PACKAGE_NAME@/g' \
 	@ $(SEDMAN) \
 	  > $@ < $<
 
+hwloc_ls_name=`echo "hwloc-ls" | $(SED) -e "$(transform)"`
+lstopo_name=`echo "lstopo" | $(SED) -e "$(transform)"`
+lstopo_no_graphics_name=`echo "lstopo-no-graphics" | $(SED) -e "$(transform)"`
+
 install-exec-hook:
-	rm -f $(DESTDIR)$(bindir)/hwloc-ls$(EXEEXT)
-	cd $(DESTDIR)$(bindir) && $(LN_S) lstopo-no-graphics$(EXEEXT) hwloc-ls$(EXEEXT)
+	rm -f $(DESTDIR)$(bindir)/$(hwloc_ls_name)$(EXEEXT)
+	cd $(DESTDIR)$(bindir) && $(LN_S) $(lstopo_no_graphics_name)$(EXEEXT) $(hwloc_ls_name)$(EXEEXT)
 if !HWLOC_HAVE_WINDOWS
 if !HWLOC_HAVE_CAIRO
-	rm -f $(DESTDIR)$(bindir)/lstopo
-	cd $(DESTDIR)$(bindir) && $(LN_S) lstopo-no-graphics$(EXEEXT) lstopo$(EXEEXT) || true
+	rm -f $(DESTDIR)$(bindir)/$(lstopo_name)$(EXEEXT)
+	cd $(DESTDIR)$(bindir) && $(LN_S) $(lstopo_no_graphics_name)$(EXEEXT) $(lstopo_name)$(EXEEXT) || true
 endif
 endif
 
 install-data-hook:
-	rm -f $(DESTDIR)$(man1dir)/hwloc-ls.1
-	cd $(DESTDIR)$(man1dir) && $(LN_S) lstopo-no-graphics.1 hwloc-ls.1
-	rm -f $(DESTDIR)$(man1dir)/lstopo.1
-	cd $(DESTDIR)$(man1dir) && $(LN_S) lstopo-no-graphics.1 lstopo.1
+	rm -f $(DESTDIR)$(man1dir)/$(hwloc_ls_name).1
+	cd $(DESTDIR)$(man1dir) && $(LN_S) $(lstopo_no_graphics_name).1 $(hwloc_ls_name).1
+	rm -f $(DESTDIR)$(man1dir)/$(lstopo_name).1
+	cd $(DESTDIR)$(man1dir) && $(LN_S) $(lstopo_no_graphics_name).1 $(lstopo_name).1
 
 uninstall-local:
-	rm -f $(DESTDIR)$(bindir)/hwloc-ls$(EXEEXT)
+	rm -f $(DESTDIR)$(bindir)/$(hwloc_ls_name)$(EXEEXT)
 if !HWLOC_HAVE_WINDOWS
 if !HWLOC_HAVE_CAIRO
-	rm -f $(DESTDIR)$(bindir)/lstopo$(EXEEXT)
+	rm -f $(DESTDIR)$(bindir)/$(lstopo_name)$(EXEEXT)
 endif
 endif
-	rm -f $(DESTDIR)$(man1dir)/hwloc-ls.1 $(DESTDIR)$(man1dir)/lstopo.1
+	rm -f $(DESTDIR)$(man1dir)/$(hwloc_ls_name).1 $(DESTDIR)$(man1dir)/$(lstopo_name).1
 
 distclean-local:
 	rm -f $(nodist_man_MANS)

--- a/utils/netloc/infiniband/Makefile.am
+++ b/utils/netloc/infiniband/Makefile.am
@@ -2,6 +2,7 @@
 # Copyright © 2013      University of Wisconsin-La Crosse.
 #                         All rights reserved.
 # Copyright © 2015-2018 Inria.  All rights reserved.
+# Copyright © 2023      Université de Reims Champagne-Ardenne.  All rights reserved.
 #
 # See COPYING in top-level directory.
 #
@@ -11,9 +12,11 @@ bin_SCRIPTS = \
         netloc_ib_gather_raw
 CLEANFILES = $(bin_SCRIPTS)
 
+netloc_ib_gather_raw_name=`echo "netloc_ib_gather_raw" | $(SED) -e "$(transform)"`
+
 install-exec-hook:
-	$(SED) -e 's/HWLOC_top_builddir\/utils\/lstopo/bindir/' -e '/HWLOC_top_builddir/d' $(DESTDIR)$(bindir)/netloc_ib_gather_raw > $(DESTDIR)$(bindir)/netloc_ib_gather_raw.tmp && mv -f $(DESTDIR)$(bindir)/netloc_ib_gather_raw.tmp $(DESTDIR)$(bindir)/netloc_ib_gather_raw
-	chmod +x $(DESTDIR)$(bindir)/netloc_ib_gather_raw
+	$(SED) -e 's/HWLOC_top_builddir\/utils\/lstopo/bindir/' -e '/HWLOC_top_builddir/d' $(DESTDIR)$(bindir)/$(netloc_ib_gather_raw_name) > $(DESTDIR)$(bindir)/$(netloc_ib_gather_raw_name).tmp && mv -f $(DESTDIR)$(bindir)/$(netloc_ib_gather_raw_name).tmp $(DESTDIR)$(bindir)/$(netloc_ib_gather_raw_name)
+	chmod +x $(DESTDIR)$(bindir)/$(netloc_ib_gather_raw_name)
 
 AM_CPPFLAGS = \
         -I$(top_builddir)/include \

